### DISCUSSION TITLE
Implementation of calculating entanglement entropy of neural wavefunctions

### DISCUSSION THREAD

**Discussion by sunwoo-kim** [netket/netket#1565] [Ideas]
Hey again,

I was wondering if there was a netket-native implementation of calculating entanglement entropy (2nd renyi, or von neumann) from the trained(or untrained) neural states.

Of course I would make the density operator by hand and diagonalise it for small system sizes, but there is a quantum monte carlo method to estimate it as well in the literature.

**Comment by gcarleo** [comment]
hi, we have been working on it, we will release the estimators open source soon

**Reply by sunwoo-kim** [reply]
Hi, Thanks for the quick reply. 

What would be the estimated timeframe? If it's like > 2 months, then I guess would just implement it myself, but if it's less than that, then I guess it wouldn't make sense to write essentially the same code, only to have it surpassed a moment later.

**Reply by gcarleo** [reply]
It is going to be less or equal to two months, realistically. @Z-Denis can give us a more precise estimate maybe

**Reply by Z-Denis** [reply]
Hi, we are submitting a paper soon and the code for computing the Rényi-2 entanglement entropy will be included in a repo. It should be in less than two months. However, unless you are dealing with VMC on a constrained Hilbert space, implementing the standard swap-operator estimator described in the literature you mentioned should be quite straightforward.

**Reply by sunwoo-kim** [reply]
Hi,

I was wondering if this feature was implemented, or will be soon(?).

**Reply by PhilipVinc** [reply]
It's already implemented in netket master for systems with discrete degrees of freedom. See https://netket.readthedocs.io/en/latest/api/_generated/experimental/observable/netket.experimental.observable.Renyi2EntanglementEntropy.html
